scripts/issue/463/stake_afrax.py

```diff
@@ -0,0 +1,50 @@
+from great_ape_safe import GreatApeSafe
+from helpers.addresses import r
+
+FRAX_AMT = 307_373e18
+# make sure we will have always enough frax for aave deposit
+EXTRA_WD = 15e18
+DAY = 86400
+
+vault = GreatApeSafe(r.badger_wallets.treasury_vault_multisig)
+trops = GreatApeSafe(r.badger_wallets.treasury_ops_multisig)
+
+afrax = vault.contract(r.treasury_tokens.aFRAX)
+
+
+def create_vault():
+    vault.init_convex()
+    vault.convex.create_vault(afrax)
+
+    vault.post_safe_tx()
+
+
+def afrax_sourcing():
+    crv3pool = trops.contract(r.treasury_tokens.crv3pool)
+    frax = trops.contract(r.treasury_tokens.FRAX)
+    dai = trops.contract(r.treasury_tokens.DAI)
+
+    trops.take_snapshot(tokens=[dai, crv3pool])
+    vault.take_snapshot(tokens=[afrax])
+
+    # swap 3crv -> frax
+    trops.init_curve()
+    trops.curve.withdraw_to_one_coin(crv3pool, FRAX_AMT + EXTRA_WD, dai)
```

Comment:
you're right, it is a bit excessive. my issue locally was more in the dai -> frax, when sim the dai -> frax, seems to lose some substantial amount on the swap of the stable value, so decided to bet safe to ensure that always the exec when posted will go thru.  (also not sure if it's my internet issues lately having funny behaviours on the local testing) 

does it not revert to you if you do the txs without any extra_wd?. since the premium will give at least ~313520 dai on the wd of crv3pool, which should do the job as per the ui. but locally it reverts for me

Reply:
withdrawal to dai is fine:
```
snapshot result for 0x042B32Ac6b453485e357938bdC38e0340d4b9276:
                    balance_before               balance_after                balance_delta
symbol                                                                                     
DAI           0.000000000000000000  313,836.828223979732934892   313,836.828223979732934892
3Crv    473,212.034618944224532502  165,839.034618944224532502  -307,373.000000000000000000
```

problem is with the swap to frax; you are not considering the surplus gained from the 3pool ppfs:
```
snapshot result for 0x042B32Ac6b453485e357938bdC38e0340d4b9276:
                    balance_before               balance_after                balance_delta
symbol                                                                                     
DAI           0.000000000000000000    6,463.828223979732934892     6,463.828223979732934892
3Crv    473,212.034618944224532502  165,839.034618944224532502  -307,373.000000000000000000
FRAX          0.000000000000000000  307,325.792947791199906191   307,325.792947791199906191
```
(the 6.4k dai left untouched here)

should be solved by doing a dusty swap, see L33.